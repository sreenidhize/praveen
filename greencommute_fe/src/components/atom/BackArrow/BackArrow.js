import React from 'react';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import { makeStyles } from "@material-ui/core/styles";

const useStyle = makeStyles((theme) => ({
    backButton: {
        display: 'flex',
        alignItems: 'center',
        cursor: "pointer"
    },
    backIcon: {
        color: theme.palette.secondary.main,
        paddingRight: theme.spacing(2),
        fontSize: theme.typography.h5.fontSize,
    },
    buttonLabel:{
        color: theme.palette.secondary.main,
        fontFamily: theme.typography.h5.fontFamily,
        fontWeight: theme.typography.h5.fontWeight,
        fontSize: theme.typography.h5.fontSize,
    }
}));

const ArrowBackComponent = (props)=>{
    const classes = useStyle();
    return (
        <div className={classes.backButton} onClick={props.onClick}>
            <ArrowBackIcon className={classes.backIcon} data-testid="arrow-icon" />
            <span className={classes.buttonLabel} data-testid="arrow-label">{props.label}</span>
        </div>
    )
}
export default ArrowBackComponent;