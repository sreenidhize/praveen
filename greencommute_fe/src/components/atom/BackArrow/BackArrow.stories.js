import React from "react";
import ArrowBackComponent from './BackArrow';

export default {
        title: "Atoms/ArrowBack",
        component: ArrowBackComponent,
        argTypes: { onClick: { action: "clicked" } },
}

const Template = (args) => (
      <ArrowBackComponent {...args} />
  );
export const Primary = Template.bind({});
  Primary.args = {
    label: "Back",
  };
export const Secondary = Template.bind({});
  Secondary.args = {
    label: " ",
  };
  
  