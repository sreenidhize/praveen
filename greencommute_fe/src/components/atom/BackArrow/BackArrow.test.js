import React from 'react';
import BackArrow from './BackArrow'
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

describe("arrow renders correctly", () => {
    test("should match snapshot", () => {
      const { asFragment, debug } = render(<BackArrow />);
      expect(asFragment()).toMatchSnapshot();
    });
    test("render back arrow",()=>{
      const{getByTestId} = render(<BackArrow label="Back Arrow" />);
      expect(getByTestId("arrow-icon")).toBeInTheDocument();
      expect(getByTestId("arrow-label")).toBeInTheDocument();
    })
});
