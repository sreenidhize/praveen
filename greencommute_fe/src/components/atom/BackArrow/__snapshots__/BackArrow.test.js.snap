// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrow renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-backButton-1"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-backIcon-2"
      data-testid="arrow-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
      />
    </svg>
    <span
      class="makeStyles-buttonLabel-3"
      data-testid="arrow-label"
    />
  </div>
</DocumentFragment>
`;
